pkg/server/plugin/zmq/rpc.go

```diff
@@ -176,6 +194,54 @@ func (p *zmqTransport) ipc(req *pluginrequest.Request) (out []byte, err error) {
 	return
 }
 
+func (p *zmqTransport) handleRequest(parcel *Parcel) {
+	workerID := parcel.Worker
+	requestID := parcel.RequestID
+	bounceCount := parcel.BounceCount
+	buffer := parcel.Frame
+	responseChan := parcel.RespChan
+
+	reader := bytes.NewReader(buffer)
+	data := map[string]interface{}{}
+	if jsonErr := json.NewDecoder(reader).Decode(&data); jsonErr != nil && jsonErr != io.EOF {
+		// TODO(b123400): handle json error
+		return
+	}
+
+	payloadData := data["payload"].(map[string]interface{})
+	method := data["method"].(string)
+	path := data["path"].(string)
+
+	payload := &router.Payload{
+		Context: context.Background(),
+		Meta: map[string]interface{}{},
+		Data: payloadData,
+		AccessKey: router.MasterAccessKey,
+	}
+
+	responseWriter := &zmqResponseWriter{}
+	resp := router.NewResponse(responseWriter)
+
+	newCtx := context.WithValue(payload.Context, ZMQWorkerIDContextKey, workerID)
+	newCtx = context.WithValue(newCtx, ZMQRequestIDContextKey, requestID)
+	newCtx = context.WithValue(newCtx, ZMQBounceCountContextKey, bounceCount)
+	payload.Context = newCtx
+
+	// TODO(b123400), put real path and method here
+	p.router.HandlePayload(path, method, payload, resp)
+
+	responseChan <- responseWriter.response
+}
+
+func (p *zmqTransport) listenRequests() {
+	for {
+		select {
+		case parcel := <- p.broker.ReqChan:
+			go p.handleRequest(parcel)
```

Comment:
IMO, unless you have other plans at mind, it's a bit odd to spawn a goroutine just to take messages from a channel and then spawn another goroutine to handle that message. Shouldn't we just spawn that goroutine at the calling site instead?